bin/improver-spot-extract

```diff
@@ -222,12 +230,13 @@ def main():
     diagnostic_names = [diagnostics[diagnostic]["filepath"]
                         for diagnostic in diagnostics.keys()]
 
+    # Create a list of files that will need to be read.
+    all_available_files = glob.glob(args.data_path+"/*")
     files_to_read = []
-    for dirpath, _, files in os.walk(args.data_path):
-        for filename in files:
-            for diagnostic_name in diagnostic_names:
-                if diagnostic_name in filename:
-                    files_to_read.append(os.path.join(dirpath, filename))
+    for filename in all_available_files:
```

Comment:
I don't know how much more inefficient this method is that the old one, do you have a sense of it? Previously there was a check in the load to ensure only the correct diagnostic was loaded, but the file list sent to is was only for the diagnostic being processed (the diagnostic name was just a fail-safe). Now you send all the files for all the different diagnostics to the load command and it is has to use iris.load to figure out what it wants.

Reply:
I'm not sure on how these options compare. I've rewritten this so that the loading is now passed a single file, rather than a list of files. Hopefully this is more efficient!

Reply:
It's now loading a list of files associated with a single diagnostic (as I read it, not a single file), which is what it previously did, so iris shouldn't have to read and interpret cube metadata only to throw it away. Thanks for making this change.